### PR TITLE
[Components with more than 1 service] LPS-185366 Refactor ModelRepository components

### PR DESCRIPTION
**Pull request by MrJohn1911**
Ticket: https://issues.liferay.com/browse/LPS-185366

### CODE REVIEW

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:stop

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by catarineaguiar** [comment]
@MrJohn1911 regression found:

```
     [exec] com.liferay.asset.search.test.AssetSearcherStagingTest > testSiteRolePermissions FAILED
     [exec]     com.liferay.portal.kernel.search.SearchException: java.lang.RuntimeException: com.liferay.portal.kernel.search.SearchException: com.liferay.portal.kernel.exception.SystemException: com.liferay.portal.kernel.exception.NoSuchGroupException: No Group exists with the primary key 46165
     [exec]     	at com.liferay.portal.kernel.search.BaseIndexer.search(BaseIndexer.java:521)
     [exec]     	at com.liferay.asset.search.test.AssetSearcherStagingTest.search(AssetSearcherStagingTest.java:210)
     [exec]     	at com.liferay.asset.search.test.AssetSearcherStagingTest.testSiteRolePermissions(AssetSearcherStagingTest.java:128)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
     [exec]     	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     [exec]     	at java.lang.reflect.Method.invoke(Method.java:498)
     [exec]     	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable$3.evaluate(TestExecutorRunnable.jav
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-05/test-1-34/test-portal-acceptance-pullrequest(master)/4099/modules-integration-mysql57-jdk8/0/3/jenkins-console.txt.gz?authuser=0)

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by catarineaguiar** [comment]
@MrJohn1911 regressions found:

```
     [exec] 
     [exec] com.liferay.asset.publisher.exportimport.test.AssetPublisherExportImportTest > testAnyDLFileEntryType FAILED
     [exec]     com.liferay.exportimport.kernel.lar.PortletDataException: com.liferay.layout.internal.model.adapter.StagedLayoutSetImpl cannot be cast to com.liferay.asset.kernel.model.AssetCategory
     [exec]     	at com.liferay.exportimport.kernel.lar.BaseStagedModelDataHandler.importStagedModel(BaseStagedModelDataHandler.java:408)
     [exec]     	at com.liferay.exportimport.kernel.lar.StagedModelDataHandlerUtil.importStagedModel(StagedModelDataHandlerUtil.java:389)
     [exec]     	at com.liferay.exportimport.kernel.lar.StagedModelDataHandlerUtil.importStagedModel(StagedModelDataHandlerUtil.java:375)
     [exec]     	at com.liferay.site.internal.exportimport.data.handler.StagedGroupStagedModelDataHandler.doImportStagedModel(StagedGroupStagedModelDataHandler.java:291)
     [exec]     	at com.liferay.site.internal.exportimport.data.handler.StagedGroupStagedModelDataHandler.doImportStagedModel(StagedGroupStagedModelDataHandler.java:74)
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-05/test-1-32/test-portal-acceptance-pullrequest(master)/9695/modules-integration-mysql57-jdk8/0/2/jenkins-console.txt.gz?authuser=0)

```
     [exec] com.liferay.fragment.staging.test.FragmentEntryStagingTest > testInputFragmentCopiedWhenLocalStagingActivated FAILED
     [exec]     java.lang.AssertionError
     [exec]     	at org.junit.Assert.fail(Assert.java:87)
     [exec]     	at org.junit.Assert.assertTrue(Assert.java:42)
     [exec]     	at org.junit.Assert.assertNotNull(Assert.java:713)
     [exec]     	at org.junit.Assert.assertNotNull(Assert.java:723)
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-05/test-1-32/test-portal-acceptance-pullrequest(master)/9695/modules-integration-mysql57-jdk8/0/21/jenkins-console.txt.gz?authuser=0)

```
     [exec] com.liferay.asset.service.test.AssetEntryLocalServiceTest > testDeleteEntry FAILED
     [exec]     java.lang.RuntimeException: com.liferay.portal.kernel.exception.NoSuchGroupException: No Group exists with the primary key 48539
     [exec]     	at com.liferay.portlet.asset.service.impl.AssetLinkLocalServiceImpl.addDeletionSystemEvent(AssetLinkLocalServiceImpl.java:443)
     [exec]     	at com.liferay.portlet.asset.service.impl.AssetLinkLocalServiceImpl.deleteAssetLink(AssetLinkLocalServiceImpl.java:118)
     [exec]     	at com.liferay.portlet.asset.service.impl.AssetLinkLocalServiceImpl.deleteLink(AssetLinkLocalServiceImpl.java:154)
     [exec]     	at com.liferay.portlet.asset.service.impl.AssetLinkLocalServiceImpl.deleteLinks(AssetLinkLocalServiceImpl.java:189)
     [exec]     	at sun.reflect.GeneratedMethodAccessor3703.invoke(Unknown Source)
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-05/test-1-32/test-portal-acceptance-pullrequest(master)/9695/modules-integration-mysql57-jdk8/0/4/jenkins-console.txt.gz?authuser=0)

```
     [exec] com.liferay.site.service.test.GroupServiceTest > testAddStagedParentToStagedGroup FAILED
     [exec]     java.lang.NullPointerException
     [exec]     	at com.liferay.site.service.test.GroupServiceTest.testAddStagedParentToStagedGroup(GroupServiceTest.java:180)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
     [exec]     	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     [exec]     	at java.lang.reflect.Method.invoke(Method.java:498)
     [exec]     	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable$3.evaluate(TestExecutorRunnable.java:353)
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-05/test-1-32/test-portal-acceptance-pullrequest(master)/9695/modules-integration-mysql57-jdk8/1/12/jenkins-console.txt.gz?authuser=0)

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:startup

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/exportimport/staged/model/repository/DDMFormInstanceRecordStagedModelRepositoryHelper.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+import java.util.Objects;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Joao Victor Alves
+ */
+@Component(service = DDMFormInstanceRecordStagedModelRepositoryHelper.class)
+public class DDMFormInstanceRecordStagedModelRepositoryHelper {
```
Comment (rafaprax) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/exportimport/staged/model/repository/DDMFormInstanceRecordStagedModelRepositoryHelper.java:38]: Hey @MrJohn1911, let's transform it into a Util class using the Snapshot approach. 

Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by Kvale1733** [comment]
@MrJohn1911 tests aren't running due to the following compile issue
```
[exec] > Task :apps:asset:asset-link-service:compileJava FAILED
     [exec] /opt/dev/projects/github/liferay-portal/modules/apps/asset/asset-link-service/src/main/java/com/liferay/asset/link/internal/exportimport/staged/model/repository/StagedAssetLinkStagedModelRepository.java:261: error: cannot find symbol
     [exec] 	private StagedAssetLinkStagedModelRepositoryHelper
     [exec] 	        ^
     [exec]   symbol:   class StagedAssetLinkStagedModelRepositoryHelper
     [exec]   location: class com.liferay.asset.link.internal.exportimport.staged.model.repository.StagedAssetLinkStagedModelRepository
     [exec] /opt/dev/projects/github/liferay-portal/modules/apps/asset/asset-link-service/src/main/java/com/liferay/asset/link/internal/exportimport/staged/model/repository/StagedAssetLinkStagedModelRepositoryUtil.java:22: error: package com.liferay.osgi.util.service does not exist
     [exec] import com.liferay.osgi.util.service.Snapshot;
     [exec]                                     ^
     [exec] /opt/dev/projects/github/liferay-portal/modules/apps/asset/asset-link-service/src/main/java/com/liferay/asset/link/internal/exportimport/staged/model/repository/StagedAssetLinkStagedModelRepositoryUtil.java:89: error: cannot find symbol
     [exec] 	private static final Snapshot<AssetLinkLocalService>
     [exec] 	                     ^
     [exec]   symbol:   class Snapshot
     [exec]   location: class com.liferay.asset.link.internal.exportimport.staged.model.repository.StagedAssetLinkStagedModelRepositoryUtil
     [exec] /opt/dev/projects/github/liferay-portal/modules/apps/asset/asset-link-service/src/main/java/com/liferay/asset/link/internal/exportimport/staged/model/repository/StagedAssetLinkStagedModelRepositoryUtil.java:93: error: cannot find symbol
     [exec] 	private static final Snapshot<StagingAssetEntryHelper>
     [exec] 	                     ^
     [exec]   symbol:   class Snapshot
     [exec]   location: class com.liferay.asset.link.internal.exportimport.staged.model.repository.StagedAssetLinkStagedModelRepositoryUtil
     [exec] /opt/dev/projects/github/liferay-portal/modules/apps/asset/asset-link-service/src/main/java/com/liferay/asset/link/internal/exportimport/staged/model/repository/StagedAssetLinkStagedModelRepositoryUtil.java:90: error: cannot find symbol
     [exec] 		_assetLinkLocalServiceSnapshot = new Snapshot<>(
     [exec] 		                                     ^
     [exec]   symbol:   class Snapshot
     [exec]   location: class com.liferay.asset.link.internal.exportimport.staged.model.repository.StagedAssetLinkStagedModelRepositoryUtil
     [exec] /opt/dev/projects/github/liferay-portal/modules/apps/asset/asset-link-service/src/main/java/com/liferay/asset/link/internal/exportimport/staged/model/repository/StagedAssetLinkStagedModelRepositoryUtil.java:94: error: cannot find symbol
     [exec] 		_stagingAssetEntryHelperSnapshot = new Snapshot<>(
     [exec] 		                                       ^
     [exec]   symbol:   class Snapshot
     [exec]   location: class com.liferay.asset.link.internal.exportimport.staged.model.repository.StagedAssetLinkStagedModelRepositoryUtil
     [exec] 6 errors
     [exec] 
```
[Log](https://testray.liferay.com/reports/production/logs/2023-05/test-1-31/test-portal-acceptance-pullrequest(master)/8148/jenkins-console.txt.gz?authuser=0)

**Pull request closed without merging**